sql/tests: fix secondary index key in sysbenchKV.DeleteInsert

nvb opened the pull request:
`sysbenchKV.DeleteInsert` was previously re-inserting the secondary index key that it had just deleted, instead of inserting the secondary index key that corresponds to the new primary index key.

To prevent this kind of bug in the future, we add some checks for secondary index integrity, where appropriate.

Epic: None
Release note: None

nvb committed:
sql/tests: fix secondary index key in sysbenchKV.DeleteInsert

`sysbenchKV.DeleteInsert` was previously re-inserting the secondary index key
that it had just deleted, instead of inserting the secondary index key that
corresponds to the new primary index key.

To prevent this kind of bug in the future, we add some checks for secondary
index integrity, where appropriate.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/137265)
<!-- Reviewable:end -->

nvb committed:
sql/tests: fix primary index value encoding

This was not working as expected, making the assertions in the next commit
impossible to pass.

Epic: None
Release note: None

tbg commented on pkg/sql/tests/sysbench_test.go:
```diff
@@ -459,36 +472,26 @@ func (s *sysbenchKV) indexKey(t tableNum, k kValue, id rowID) []byte {
 }
 
 func (s *sysbenchKV) encodePKValue(b []byte, k kValue) {
 	// Replace the first 8 bytes with the k value, so that we can decode k to keep
```
The comment is stale.

tbg replied:
Actually it's not, nevermind.
<!-- Reviewable comment -ODu1iyt4oqJrnB2QwzY:b-saj7te -->
<!-- Sent from Reviewable.io -->

tbg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137265#-:-ODu23d49N26i4Shyrxa:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten and @RaduBerinde)

___
*[`pkg/sql/tests/sysbench_test.go` line 478 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137265#-ODu1aXC0O1TzUVB52Ca:-ODu1aXC0O1TzUVB52Cb:b-ytww5s) ([raw file](https://github.com/cockroachdb/cockroach/blob/bfdabf4fdde5ebbd9af2f6493682ca79abaeed13/pkg/sql/tests/sysbench_test.go#L478)):*
> ```Go
> 
> func (s *sysbenchKV) decodePKValue(b []byte) kValue {
> 	// Extract the k value from the first 8 bytes. The rest is random.
> ```

Possibly also merits an update to the comment.


<!-- Sent from Reviewable.io -->

nvanbenschoten commented:
TFTR!

bors r+
